src/EditorFeatures/Core/Implementation/IntelliSense/SignatureHelp/Presentation/Signature.cs

```diff
@@ -16,26 +17,99 @@ internal class Signature : ISignature
         private const int MaxParamColumnCount = 100;
 
         private readonly SignatureHelpItem _signatureHelpItem;
-        private IParameter _currentParameter;
-        internal IList<SymbolDisplayPart> DisplayParts { get; private set; }
+
+        public Signature(ITrackingSpan applicableToSpan, SignatureHelpItem signatureHelpItem, int selectedParameterIndex)
+        {
+            this.ApplicableToSpan = applicableToSpan;
+            _signatureHelpItem = signatureHelpItem;
+            _parameterIndex = selectedParameterIndex;
+        }
+
+        private bool _isInitialized;
```

Comment:
Don't get me started on Lazy<T> :-)
A Lazy<T> is a reference type (i.e. an allocation) and the valueFactory Func<T> will, in this case, be a closure over 'this' (another allocation).

These things are created like candy during typing scenarios. For example, when you're typing Console.WriteLine("Hello, World!") each character typed within the parentheses will create (at least) 19 instances - one for each overload of WriteLine. In this case, I think the extra typing is justified.